server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -53,12 +53,18 @@
     private List<File> gradlePackages;
     private int nameCounter = 0;
 
+
+    static final String FOLDER_ENGINE_SRC = "src";      // source for the engine library
+    static final String FOLDER_PLUGIN_SRC = "pluginsrc";// source for the pipeline/format plugins
+    static final String FOLDER_COMMON_SRC = "commonsrc";// common source shared between both types
```

Comment:
These names are now part of the api.
Are they good enough?

Reply:
Is plugin the word we are going to use? When talking to users or when users discuss Defold I see the following three used to describe what we call a Library project in the documentation:

* (Native) extension
* library
* plugin
* dependency

And soon an extension project will also contain source code for the editor right? That folder would probably be named `editorsrc`?

What about `buildsrc`, `buildersrc` or `pipelinesrc`?

`commonsrc` is good I think. It could also be named `sharedsrc` but I believe we use `common` elsewhere to indicate shared things.

Reply:
"plugin" was a collecting term for all plugins (bob _and_ editor)

"buildsrc" is too ambigous I think. All source is built.

"pipelinesrc" is better.

I don't think we'll want a separate "editorsrc". I consider that part still a "plugin"

Reply:
I think it's good to separate the meanings of the the terms (as we already do).
I wouldn't want to call everything "extensions" or "libraries".

Also, I don't really consider them confusing.
But I agree that we should be more specific when referring to them

* (Defold) library
* Dependency is the link to a library
* (Native) extension - statically linked source code. Avaialble through the Defold Library system.
* Plugin (a separate file/bundle to add code)

Reply:
I chose "pluginsrc" over "pipelinesrc" because it was shorter, but more importantly because "plugin" is a well known concept. But as you mention, it might be confusing for others.